lib/mail/field.rb

```diff
@@ -232,12 +232,13 @@ def split(raw_field)
     #  that is immediately followed by WSP.  Each header field should be
     #  treated in its unfolded form for further syntactic and semantic
     #  evaluation.
-    def unfold(string)
-      string.gsub(/[\r\n \t]+/m, ' ')
+    def unfold(name, value)
+      regex = name == 'Subject' ? /\s*\r\n\s+/m : /[\r\n \t]+/m
```

Comment:
@kjg That was my original approach, but based on existing specs, there are some header fields where the whitespace is supposed to be stripped out. I wasn't able to come up with a single regex that passed both my new specs and all existing specs.